devito/types/tensor.py

```diff
@@ -79,211 +125,89 @@ def __subfunc_setup__(cls, *args, **kwargs):
             funcs = funcs.tolist()
         return funcs
 
-    def __getattr__(self, name):
+    @call_highest_priority('__mul__')
+    def __rmul__(self, other):
         """
-        Try calling a dynamically created FD shortcut.
-
-        Notes
-        -----
-        This method acts as a fallback for __getattribute__
+        Prevents Functions to be interpreted as matrices in 2D becaue of `.shape`
```

Comment:
because

Reply:
because sympy assumes that everything with a `.shape` is a matrix but out Function have a `shape` that is not compatible and should not be processed as a "matrix shape". Basically fucks up sympy, that;s what the `is_Matrix=Flase, is_MatrixLike=False` are for but will only be processed properly in 1.7